syntax = "proto3";

package game;

service Network {
    rpc Register(Void) returns (OrderInfo) {}

    rpc SendEvent(Event) returns (Void) {}

    rpc GetEvents(Player) returns (stream Event) {}
}

message Void {}

message OrderInfo {
    int32 id = 1;
    int32 numberOfPlayers = 2;
}

enum EventType {
    CARD = 0;
    DICE = 1;
    MARKET = 2;
    BUILD = 3;
    ENDTURN = 4;
    //TRADE = 5;
}

message Event {
    EventType type = 1;
    Player player = 2;
    oneof EventInfo {
        Card cardInfo = 3;
        Dice diceInfo = 4;
        Market marketInfo = 5;
        Build buildInfo = 6;
    }
}

message Card {
    int32 cardType = 1;
}

message Dice {
    int32 number = 1;
}

message Market {
    int32 requiredResource = 1;
    int32 ownedResource = 2;
}

message Build {
    int32 x = 1;
    int32 y = 2;
}

message Player {
    int32 playerId = 1;
}


